pyxrf/model/load_data_from_db.py

```diff
@@ -347,6 +354,128 @@ def _is_scan_complete(hdr):
+        #   No locations means that the data is not yet defined in start document on any beamline
+        #   Multiple locations point to locations at different beamlines
+        if not locations:
+            continue
+
+        # For each metadata key there could be none, one or multiple locations in the start document
+        for loc in locations:
+            path = loc.split('/')  #
+            ref = start_document
+            for n, p in enumerate(path):
+                if n >= len(path) - 1:
+                    break
+                # 'ref' must always point to dictionary
+                if not isinstance(ref, dict):
+                    ref = None
+                    break
+                if p in ref:
+                    ref = ref[p]
+                else:
+                    ref = None
+                    break
+            # At this point 'ref' must be a dictionary
+            value = None
+            if ref is not None and isinstance(ref, dict):
+                if path[-1] in ref:
+                    value = ref[path[-1]]
+            # Now we finally arrived to the end of the path: the 'value' must be a scalar or a list
+            if value is not None and not isinstance(value, dict):
+                if path[-1] == 'time':
+                    if key.endswith("_utc"):
+                        value = convert_time_to_nexus_string(ttime.gmtime(value))
+                    else:
+                        value = convert_time_to_nexus_string(ttime.localtime(value))
+                mdata[key] = value
+                break
+
+    stop_document = hdr.stop
+
+    if stop_document:
+
+        if "time" in stop_document:
+            t = stop_document["time"]
+            mdata["scan_time_stop"] = convert_time_to_nexus_string(ttime.localtime(t))
+            mdata["scan_time_stop_utc"] = convert_time_to_nexus_string(ttime.gmtime(t))
+
+        if "exit_status" in stop_document:
+            mdata["scan_exit_status"] = stop_document["exit_status"]
+
+    else:
+
+        mdata["scan_exit_status"] = "incomplete"
+
+    # Add full beamline name (if available, otherwise don't create the entry).
+    #   Also, don't overwrite the existing name if it was read from the start document
+    if "scan_instrument_id" in mdata and "scan_instrument_name" not in mdata:
+        instruments = {
+            "srx": "Submicron Resolution X-ray Spectroscopy",
+            "hxn": "Hard X-ray Nanoprobe",
+            "tes": "Tender Energy X-ray Absorption Spectroscopy",
+            "xfm": "X-ray Fluorescence Microprobe"
+        }
+        iname = instruments.get(mdata["scan_instrument_id"].lower(), "")
+        if iname:
+            mdata["scan_instrument_name"] = iname
+
+    return mdata
+
+
+def _get_metadata_from_descriptor_document(hdr, *, data_key, stream_name='baseline'):
+
+    # Returns None if the parameter is not found
+
+    value = None
+    docs = hdr.documents(stream_name=stream_name)
+    for doc in docs:
```

Comment:
We normally loop over it as `for name, doc in docs: ...`. This way you don't have to access the 0th and 1st elements of the returned pair, which is done below.

Reply:
Fixed